drivers/at86rf2xx/at86rf2xx_netdev.c

```diff
@@ -433,7 +433,7 @@ static int _set(netdev2_t *netdev, netopt_t opt, void *val, size_t len)
                 uint8_t chan = ((uint8_t *)val)[0];
                 if (chan < AT86RF2XX_MIN_CHANNEL ||
                     chan > AT86RF2XX_MAX_CHANNEL) {
-                    res = -ENOTSUP;
+                    res = -EINVAL;
                     break;
                 }
                 at86rf2xx_set_chan(dev, chan);
```

Comment:
maybe I am confused, but something still seems off: initially, `res` is set to `-ENOTSUP`. If the given channel is invalid, we set it to `-EINVAL`, so far so good. But if the channel is valid, we do not change `res`, so when we get to the break below, `res` is still set to `-ENOTSUP`. This will trigger the call to `netdev2_ieee802154_set(...)` at the end of the function. Is this intentional?